src/Core__Type.res

```diff
@@ -1,43 +1,109 @@
-  external _asSymbol: 'a => symbol = "%identity"
-
-  let classify = value => {
-    switch _internalClass(value) {
-    | "[object Boolean]" => Bool(_asBool(value))
-    | "[object Null]" => Null
-    | "[object Undefined]" => Undefined
-    | "[object String]" => String(_asString(value))
-    | "[object Number]" => Number(_asFloat(value))
-    | "[object Function]"
-    | "[object GeneratorFunction]"
-    | "[object AsyncFunction]" =>
-      Function(_asFunction(value))
-    | "[object Symbol]" => Symbol(_asSymbol(value))
-    | _ => Object(_asObject(value))
+type typeof = [
+  | #undefined
+  | #object
+  | #boolean
+  | #number
+  | #bigint
+  | #string
+  | #symbol
+  | #function
+]
+
+external typeof: 'a => typeof = "#typeof"
+
+type unknown
+external toUnknown: 'a => unknown = "%identity"
+
+type object // How to use with the Object module?
+type function
+
+// typeof null == "object". Gotta love that! Do better here.
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof#typeof_null
+type jsType =
+  | Undefined
+  | Null
+  | Object(object)
+  | Bool(bool)
+  | Number(float)
+  | BigInt(Core__BigInt.t)
+  | String(string)
+  | Symbol(Core__Symbol.t)
+  | Function(function)
+
+external toObjectUnsafe: 'a => object = "%identity"
+external toBoolUnsafe: 'a => bool = "%identity"
+external toFloatUnsafe: 'a => float = "%identity"
+external toBigIntUnsafe: 'a => Core__BigInt.t = "%identity"
+external toStringUnsafe: 'a => string = "%identity"
+external toSymbolUnsafe: 'a => Core__Symbol.t = "%identity"
+external toFunctionUnsafe: 'a => function = "%identity"
+
+let isNull = %raw(`function(a) { return (a===null); }`)
+let isNullOrUndefined = %raw(`function(a) { return (a===null || a===undefined); }`)
+let isUndefined = %raw(`function(a) { return (a===undefined); }`)
+
+let classify = value => {
+  switch typeof(value) {
+  | #number => value->toFloatUnsafe->Number
+  | #string => value->toStringUnsafe->String
+  | #boolean => value->toBoolUnsafe->Bool
+  | #undefined => Undefined
+  | #function => value->toFunctionUnsafe->Function
+  | #bigint => value->toBigIntUnsafe->BigInt
+  | #symbol => value->toSymbolUnsafe->Symbol
+  | #object =>
+    switch isNull(value) {
+    | true => Null
+    | false => value->toObjectUnsafe->Object
     }
   }
 }
+
+let toObject = i =>
+  switch i->classify {
+  | Object(i) => Some(i)
+  | _ => None
+  }
+
+let toBool = i =>
+  switch i->classify {
+  | Bool(b) => Some(b)
+  | _ => None
+  }
+
+let toFloat = i =>
+  switch i->classify {
+  | Number(i) => Some(i)
+  | _ => None
+  }
+
+let toBigInt = i =>
+  switch i->classify {
+  | BigInt(i) => Some(i)
+  | _ => None
+  }
+
+let toString = i =>
+  switch i->classify {
+  | String(i) => Some(i)
+  | _ => None
+  }
+
+let toSymbol = i =>
+  switch i->classify {
+  | Symbol(i) => Some(i)
+  | _ => None
+  }
+
+let toFunction = i =>
+  switch i->classify {
+  | Function(i) => Some(i)
+  | _ => None
+  }
```

Comment:
These can be optimized by using `typeof` directly

```suggestion
let toFunction = value => typeof(value) == #function ? value->Obj.magic->Some : None
```

Also, `i` is a really weird variable name to use for this.

Reply:
Ok did that